esp/esdllib/esdl_script.cpp

```diff
@@ -771,111 +829,129 @@ class CEsdlTransformOperationHttpPostXml : public CEsdlTransformOperationBase
+                throw MakeStringException(ESDL_SCRIPT_Error, "ESDL Script error sending request in http-post-xml %s url(%s)", m_traceName.str(), preparedState.url.str());
 
             resp->getContent(response);
             if (!response.trim().length())
-                throw MakeStringException(ESDL_SCRIPT_Error, "ESDL Script empty result calling http-post-xml %s url(%s)", m_traceName.str(), url);
+                throw MakeStringException(ESDL_SCRIPT_Error, "ESDL Script empty result calling http-post-xml %s url(%s)", m_traceName.str(), preparedState.url.str());
 
-            StringBuffer contentType;
+            resp->getStatus(status);
             resp->getContentType(contentType);
-            targetContext->ensureSetValue("@content-type", contentType.str(), true);
-            if (strnicmp("text/xml", contentType.str(), 8)==0 || strnicmp("application/xml", contentType.str(), 15) ==0)
-            {
-                CXpathContextLocation content_location(targetContext);
-                targetContext->addElementToLocation("content");
-                targetContext->addXmlContent(response.str());
-            }
-            else
-            {
-                targetContext->ensureSetValue("@status", "error", true);
-                targetContext->ensureSetValue("text", response.str(), true);
-            }
         }
         catch(IMultiException *me)
         {
-            StringBuffer xml;
-            me->serialize(xml);
-            CXpathContextLocation content_location(targetContext);
-            targetContext->ensureSetValue("@status", "error", true);
-            targetContext->addElementToLocation("content");
-            targetContext->addXmlContent(xml.str());
+            me->serialize(exceptionXml);
             me->Release();
         }
         catch(IException *E)
         {
-            StringBuffer xml;
             Owned<IMultiException> me = makeMultiException("ESDLScript");
             me->append(*LINK(E));
-            me->serialize(xml);
-            CXpathContextLocation content_location(targetContext);
-            targetContext->ensureSetValue("@status", "error", true);
-            targetContext->addElementToLocation("content");
-            targetContext->addXmlContent(xml.str());
+            me->serialize(exceptionXml);
             E->Release();
         }
-    }
 
-    virtual bool process(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext) override
-    {
-        VStringBuffer xpath("/esdl_script_context/%s/%s", m_section.str(), m_name.str());
+        if (preparedState.testDelay)
+            MilliSleep(preparedState.testDelay);
+
+        //No need to synchronize until we hit this point and start updating the scriptContext / xml document
+        OptionalCriticalBlock block(crit);
+
         CXpathContextLocation location(targetContext);
+        VStringBuffer xpath("/esdl_script_context/%s/%s", m_section.str(), m_name.str());
```

Comment:
trivial: Could be before the critical block

Reply:
done.